drivers/pwm/pwm_stm32.c

```diff
@@ -660,27 +676,36 @@ static void pwm_stm32_irq_config_func_##index(const struct device *dev)        \
 replaced by 'st,prescaler' property in parent node, aka timers"
 #endif
 
-#define PWM_DEVICE_INIT(index)                                                 \
-	static struct pwm_stm32_data pwm_stm32_data_##index;                   \
-	IRQ_CONFIG_FUNC(index)						       \
-									       \
-	PINCTRL_DT_INST_DEFINE(index);					       \
-									       \
-	static const struct pwm_stm32_config pwm_stm32_config_##index = {      \
-		.timer = (TIM_TypeDef *)DT_REG_ADDR(DT_INST_PARENT(index)),    \
-		/* For compatibility reason, use pwm st_prescaler property  */ \
-		/* if exist, otherwise use parent (timers) property         */ \
-		.prescaler = DT_INST_PROP_OR(index, st_prescaler,              \
-			(DT_PROP(DT_INST_PARENT(index), st_prescaler))),       \
-		.pclken = DT_INST_CLK(index, timer),                           \
-		.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(index),		       \
-		CAPTURE_INIT(index)					       \
-	};                                                                     \
-									       \
-	DEVICE_DT_INST_DEFINE(index, &pwm_stm32_init, NULL,                    \
-			    &pwm_stm32_data_##index,                           \
-			    &pwm_stm32_config_##index, POST_KERNEL,            \
-			    CONFIG_KERNEL_INIT_PRIORITY_DEVICE,                \
+#define DT_TIMER_COUNTER_MODE(inst)                                                  \
+	(DT_ENUM_IDX(inst, countermode) == 0) ? LL_TIM_COUNTERMODE_UP                \
+	: ((DT_ENUM_IDX(inst, countermode) == 1) ? LL_TIM_COUNTERMODE_DOWN           \
+	: ((DT_ENUM_IDX(inst, countermode) == 2) ? LL_TIM_COUNTERMODE_CENTER_DOWN    \
+	: ((DT_ENUM_IDX(inst, countermode) == 3) ? LL_TIM_COUNTERMODE_CENTER_UP      \
```

Comment:
Suggestion: you can also use an `int` field in DT and provide definitions via `include/dt-bindings/...`, e.g.

```
countermode = <STM32_TIM_COUNTERMODE_CENTER_UP>;
```

its value would be equal to `LL_...`.

Reply:
It doesn't look like I can include `stm32_ll_tim.h` in the dt-bindings header.
But that is not a problem as the `CMS` value is similar between most SoCs.